pkg/controller/kubelet-config/helpers.go

```diff
@@ -67,10 +67,14 @@ func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool) string {
 
 // validates a KubeletConfig and returns an error if invalid
 func validateUserKubeletConfig(cfg *mcfgv1.KubeletConfig) error {
-	if cfg.Spec.KubeletConfig == nil {
+	if cfg.Spec.KubeletConfig.Raw == nil {
 		return nil
 	}
-	kcValues := reflect.ValueOf(*cfg.Spec.KubeletConfig)
+	kcDecoded, err := decodeKubeletConfig(cfg.Spec.KubeletConfig.Raw)
+	if err != nil {
+		return fmt.Errorf("KubeletConfig could not be unmarshalled, err: %v", err)
+	}
+	kcValues := reflect.ValueOf(*kcDecoded)
```

Comment:
the banned field logic below works, but it is totally untyped. Would prefer to have real types here, especially because field names might change from config version to config version.

Reply:
I don't think I changed the logic here. (kcDecoded is still typed as KubeletConfiguration as before). Do you mean to put a type around [blacklistKubeletConfigurationFields](https://github.com/LorbusChris/machine-config-operator/blob/master/pkg/controller/kubelet-config/kubelet_config_controller.go#L62)? Do you have an example for guidance at hand?

Reply:
no, you did not change that. But with RawExtension it's getting harder to reason about.

A good pattern is to type-cast to a real type after decoding. And then compare the real fields, not via reflection.

Reply:
> A good pattern is to type-cast to a real type after decoding.

decode routine already type-cast to `KubeletConfiguration`. Are you suggesting instead of playing with reflections below, do something like:

```go
// basically remove the loop and go one by one for banned fields
if kcDecoded.CgroupDriver != "" {
    return errors.New("CgroupDriver is not allowed to be set")
}
...
```

Reply:
resolved in https://github.com/openshift/machine-config-operator/pull/923